homeassistant/components/tradfri/base_class.py

```diff
@@ -0,0 +1,96 @@
+"""Base class for IKEA TRADFRI."""
+import logging
+
+from pytradfri.error import PytradfriError
+
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from . import DOMAIN as TRADFRI_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TradfriBaseDevice(Entity):
+    """Base class for a TRADFRI device."""
+
+    def __init__(self, device, api, gateway_id):
+        """Initialize a device."""
+        self._available = True
+        self._api = api
+        self._device = None
+        self._device_control = None
+        self._device_data = None
+        self._gateway_id = gateway_id
+        self._name = None
+        self._unique_id = None
+
+        self._refresh(device)
+
+    @callback
+    def _async_start_observe(self, exc=None):
+        """Start observation of device."""
+        if exc:
+            self._available = False
+            self.async_schedule_update_ha_state()
+            _LOGGER.warning("Observation failed for %s", self._name, exc_info=exc)
+
+        try:
+            cmd = self._device.observe(
+                callback=self._observe_update,
+                err_callback=self._async_start_observe,
+                duration=0,
+            )
+            self.hass.async_create_task(self._api(cmd))
+        except PytradfriError as err:
```

Comment:
What can raise here ? You are using create task so you're not waiting for the result.